FuelGaugeReader/FuelGaugeReader.ino

```diff
@@ -0,0 +1,38 @@
+#define USE_USBCON
+#include <ros.h>
+#include <std_msgs/Int8.h>
+std_msgs::Int8 percent_full;
+ros::Publisher pub_percent_full("percent_full", &percent_full);
+ros::NodeHandle nh;
+
+void setup()
+{
+  nh.initNode();
+  Serial.begin(9600);
+  nh.getHardware()->setBaud(9600);
+  nh.advertise(pub_percent_full);
+}
+
+long publisher_timer;
+const int READ_PIN = A7;
+
+void loop() {
+  if (millis() > publisher_timer) {
+    // outputs range from ~494 at empty to ~714 at full
+    // so shift that to 0-220 and scale to 0-100
+    percent_full.data = ((analogRead(READ_PIN) - 494) * 100 / 220);
```

Comment:
Lets do this a little different. Make it more system agnostic. + we don't like magic numbers that come out of no where. comments are okay but variable names that are defined at the top are better. Especially if we change sensors. 

Please define the following. 
analog reference voltage ( MCU dependent you are probably 5V, teensy is 3.3, internal ref pins are 1.5V)
adc resolution - 10bit adc is likely 1024 - this defines the max adc value you will see

Use the above to convert to input voltage. 
fuel_empty_voltage
fuel_full_voltage

use the above to get the percentage.

Reply:
Fixed